### PR TITLE
use capabilities on the v1 api

### PR DESCRIPTION
**Pull request by mwhudson**
This actually makes the v1 and v2 guided apis a bit closer, which is nice. This area of the code is not very well tested though!

Needs a rebase but going to stick it up for review / comment anyway.

### CODE REVIEW

File: subiquity/common/types.py
Comment (dbungert) [inline subiquity/common/types.py:322]: Are these `prefer` capabilities staying or going, eventually?

File: subiquity/ui/views/filesystem/guided.py
Comment (dbungert) [inline subiquity/ui/views/filesystem/guided.py:319]: Double check this one

File: subiquity/server/controllers/filesystem.py
Comment (dbungert) [inline subiquity/server/controllers/filesystem.py:377]: would we also want use_tpm == True for PREFER_ENCRYPTED?

File: subiquity/ui/views/filesystem/guided.py
Comment (mwhudson) [reply subiquity/ui/views/filesystem/guided.py:319]: thanks :)

File: subiquity/common/types.py
Comment (mwhudson) [reply subiquity/common/types.py:322]: We talked about this I think? They will probably go eventually and the list of capabilities end up being replaced by something richer that indicates availability and default.

**Review by dbungert — APPROVED** [review]

**Comment by dbungert** [comment]
Friendly reminder to rebase before merge button :)

**Pull request merged**
